engine/rpcserver_test.go

```diff
@@ -4139,3 +4148,99 @@ func TestGetOpenInterest(t *testing.T) {
 	_, err = s.GetOpenInterest(context.Background(), req)
 	assert.NoError(t, err)
 }
+
+func TestStartRPCRESTProxy(t *testing.T) {
+	t.Parallel()
+
+	gRPCPort := rand.Intn(65535-42069) + 42069 //nolint:gosec // Don't require crypto/rand usage here
+	gRPCProxyPort := gRPCPort + 1
+
+	e := &Engine{
+		Config: &config.Config{
+			RemoteControl: config.RemoteControlConfig{
+				Username: "bobmarley",
+				Password: "Sup3rdup3rS3cr3t",
+				GRPC: config.GRPCConfig{
+					Enabled:                true,
+					ListenAddress:          "localhost:" + strconv.Itoa(gRPCPort),
+					GRPCProxyListenAddress: "localhost:" + strconv.Itoa(gRPCProxyPort),
+				},
+			},
+		},
+		Settings: Settings{
+			DataDir:      common.GetDefaultDataDir(""),
+			CoreSettings: CoreSettings{EnableGRPCProxy: true},
+		},
+	}
+
+	fakeTime := time.Now().Add(-time.Hour)
+	e.uptime = fakeTime
+
+	StartRPCServer(e)
+
+	// Give the proxy time to start
+	time.Sleep(time.Millisecond * 500)
+
+	targetDir := utils.GetTLSDir(common.GetDefaultDataDir(""))
```

Comment:
How about a temp dir instead of going with a user's files?
```go
tempDir := filepath.Join(os.TempDir(), "gct-temp-tls")
	cleanup := func() {
		if err := os.RemoveAll(tempDir); err != nil {
			t.Errorf("unable to remove temp dir %s, manual deletion required", tempDir)
		}
	}
```

Reply:
Yup, everything else was mocked so this can be too 👍